src/users/usersActions.js

```diff
@@ -43,27 +113,23 @@ export const sendTypingStart = (narrow: Narrow) => async (
     return;
   }
 
-  const nopWorker = {
-    get_recipient: () => undefined,
-    is_valid_conversation: () => true,
-    get_current_time: () => new Date().getTime(),
-    notify_server_start: () => {},
-    notify_server_stop: () => {},
-  };
-  typing_status.handle_text_input(nopWorker);
-
-  const now = new Date();
-  if (differenceInSeconds(now, lastTypingStart) < 15) {
-    // TODO throttle properly -- e.g. if typing furiously for 14s,
-    //      don't treat as if just typed briefly at start
-    return;
-  }
-  lastTypingStart = now;
+  const usersByEmail = getAllUsersByEmail(getState());
+  const recipientIds = caseNarrowPartial(narrow, {
+    pm: email => [email],
+    groupPm: emails => emails,
+  }).map(email => {
+    const user = usersByEmail.get(email);
+    if (!user) {
+      throw new Error('unknown user');
+    }
+    return user.user_id;
+  });
 
   const auth = getAuth(getState());
-  api.typing(auth, narrow[0].operand, 'start');
+  maybeNotifyTyping(auth, recipientIds);
 };
 
+// TODO call this on more than send: blur, navigate away, etc.
```

Comment:
In "etc." perhaps include "user deletes all unsent text".

Reply:
Sure.